fine_tuning_embeddings/fine_tune_embed_model.py

```diff
@@ -0,0 +1,499 @@
+    # be sure to use path output_data_directory + train_nodes.pkl and val_nodes.pkl to check if the files exist
+    if os.path.exists(
+        args.output_data_directory + "/" + "train_nodes.pkl"
+    ) and os.path.exists(args.output_data_directory + "/" + "val_nodes.pkl"):
+        logging.info("Loading nodes from .pkl files.")
+        train_nodes_pkl = load_nodes_from_pkl(
+            args.output_data_directory + "/" + "train_nodes.pkl"
+        )
+        val_nodes_pkl = load_nodes_from_pkl(
+            args.output_data_directory + "/" + "val_nodes.pkl"
+        )
+    elif args.train_nodes_pkl_file is None or args.val_nodes_pkl_file is None:
+        train_nodes_pkl, val_nodes_pkl = generate_corpus(
+            args.input_data_directory,
+            args.output_data_directory,
+            args.file_extension,
+            args.split_ratio,
+        )
+
+    else:
+        train_nodes_pkl = load_nodes_from_pkl(args.train_nodes_pkl_file)
+        val_nodes_pkl = load_nodes_from_pkl(args.val_nodes_pkl_file)
+
+    # Instantiate LLM
+    llm = instantiate_llm()
+
+    # check if the train_dataset_path and val_dataset_path exists, if they do, load them
+
+    if os.path.exists(
+        args.output_data_directory + "/" + "train_dataset.json"
+    ) and os.path.exists(args.output_data_directory + "/" + "val_dataset.json"):
+        logging.info("Loading datasets from .json files.")
+        train_dataset_path = args.output_data_directory + "/" + "train_dataset.json"
+        val_dataset_path = args.output_data_directory + "/" + "val_dataset.json"
+        logging.info(
+            f"Train dataset path: {train_dataset_path}, Val dataset path: {val_dataset_path}"
+        )
+
+    elif args.train_dataset_path is None or args.val_dataset_path is None:
+        # Generate datasets using loaded nodes
+        train_dataset_generator = DatasetGenerator(llm=llm, nodes=train_nodes_pkl)
+        val_dataset_generator = DatasetGenerator(llm=llm, nodes=val_nodes_pkl)
+        train_dataset = train_dataset_generator.generate_dataset()
+        val_dataset = val_dataset_generator.generate_dataset()
+
+        # Save datasets
+        train_dataset_path = os.path.join(
+            args.output_data_directory, "train_dataset.json"
+        )
+        val_dataset_path = os.path.join(args.output_data_directory, "val_dataset.json")
+        train_dataset.save_json(train_dataset_path)
+        val_dataset.save_json(val_dataset_path)
+    else:
+        train_dataset_path = args.train_dataset_path
+        val_dataset_path = args.val_dataset_path
+
+    return train_dataset_path, val_dataset_path
+
+
+def main():
+    args = parse_arguments()
+
+    train_csv, val_csv = generate_data(args)
+
+    # Pass force_retrain flag to the finetune function
+    finetune(
+        train_csv, val_csv, args.model_id, args.model_output_path, args.force_retrain
+    )
+
+    openai_embedding = OpenAIEmbedding()
```

Comment:
Is this required? It will be great if we can remove any dependency on OpenAI apis.

Reply:
Not required, but optional, now just using sentence transformers comparisons